Add new control plane handler

c-p-i-o opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #129712

Summary:
Add a new control plane handler to retrieve flight recorder data as
JSON.

Test Plan:
Unit tests

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @chauhang @d4l3k

c-p-i-o committed:
Update

[ghstack-poisoned]

c-p-i-o committed:
Update

[ghstack-poisoned]

wconstab commented on torch/csrc/distributed/c10d/NCCLUtils.cpp:
```diff
@@ -289,6 +289,51 @@ control_plane::RegisterHandler dumpHandler{
+      const std::string includeCollectivesStr = "includecollectives";
+      const std::string onlyActiveStr = "onlyactive";
+
+      std::unordered_map<std::string, bool> expectedParams = {
```
nit: maybe name 'processedParams' or something so its clear that this is reused for defaults and merged with user inputs?

c-p-i-o committed:
Update on "Add new control plane handler"


Summary:
Add a new control plane handler to retrieve flight recorder data as
JSON.

Test Plan:
Unit tests

cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab chauhang d4l3k

[ghstack-poisoned]

c-p-i-o committed:
Update

[ghstack-poisoned]

c-p-i-o commented:
@pytorchbot merge

c-p-i-o commented:
@pytorchbot merge
